### ISSUE TITLE
Take interpolation code from wavefronts.py and put into a script in /utils

### ISSUE THREAD

**Issue by LouisDesdoigts** [LouisDesdoigts/dLux#71]
Similar as to how the paraxial coordinate arrays have now been branched off into their own `coordinates.py` script, we should also do the same for the interpolation code. The actual interpolation is general and is mostly just doing a little bit of fancy footwork to re-cast basic scipy interpolation functionality to work with the jax implemented `jax.scipy.ndimage.map_coordinates`. I have built basic code to do paraxial rotations and am in the processes of doing the same for the focal-plane distortions applied to JWST images. This functionality could be very useful for re-sizing and rotating apertures that are loaded from a file rather than generated parametrically. Maybe in the long term we contribute these to jax, since really all we are doing is replicating the behavior of the scipy functions.

**Comment by Jordan-Dennis** [comment]
I also have rotation code, although mine requires `xy` coordinates in a tensor. We should probably try and only have one implementation of one thing. To save you searching mine is:
```py
rotation_matrix = np.array([
    [np.cos(theta), -np.sin(theta)], 
    [np.sin(theta), mp.cos(theta)]])
return np.apply_along_axis(matmul, xy, rotation_matrix)
```
Reply with yours so that we can decide which to keep. I think these should live in `coordinates.py`. I also have magnifications, translations and shears that it might be worth considering moving here as well.

**Comment by LouisDesdoigts** [comment]
My code is more specific towards interpolation. I just convert to into polar units and add the rotation, which doesn't require a matmul. The difference between the two is *probably* trivial but I guess I can see cases where one might beat out the other on speed (depending on array sizes), since I *think* converting to polar is simply applying element-wise operations.

```def paraxial_rotate(array, rotation, order=1):
    """
    array: Array to rotate
    rotation: rotation angle in degrees
    order: interpolation order, supports 0 and 1
    
    Basically a jax version of scipy.ndimage.rotate - Contribute to Jax?
    """

    # Get coords arrays
    npix = array.shape[0]
    centre = (npix-1)/2
    x_pixels, y_pixels = get_pixel_positions(npix)
    rs, phis = cart2polar(x_pixels, y_pixels)
    phis += deg2rad(rotation)
    coordinates_rot = np.roll(polar2cart(rs, phis) + centre, shift=1, axis=0)
    rotated = map_coordinates(array, coordinates_rot, order=order)
    return rotated```

**Comment by Jordan-Dennis** [comment]
Your actually look nicer, I haven't seen `np.roll` before. I might keep mine separate for now because it is used in a stack of other operations and needs to occur after some others.

**Comment by LouisDesdoigts** [comment]
Yeah your method is definitely the way to go for aperture vertex rotations and such, and I think mine handles large coordinate arrays better. They both have their own use-case so it's still probably worth coalating them somewhere in the long run.

**Comment by Jordan-Dennis** [comment]
Hello again,
So I have been using this, this morning and I think it could quite safely be its own layer. In fact I would vouch for this but the layer may use a `util` function.

**Comment by LouisDesdoigts** [comment]
Done in 0.9.
